verify source integrity in workflows

plowsof opened the pull request:
Closes #1576 

[Help for reviewer](https://github.com/plowsof/validate-sources)

erciccione reviewed (changes requested):
Please improve the commit message to something clearer. Like `workflows: add source integrity validation to hashes.yml`.

plowsof commented:
Thanks for that, i've changed the commit message, and removed some extra white spaces i was using to compare the git clone lines.  Please bare with me on these git nuances, i created a patch on my master branch just yesterday, oops!
